src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -51,58 +51,89 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
         // Import records in Active_Fields__c, while Batches created in geBatchGiftEntryApp
-        // use all or a subset of fields mapped in Advanced Mapping.
+        // use fields included in the Form_Template__c record related to the Batch.
         DataImportBatch__c batch = [
-                SELECT Batch_Gift_Entry_Version__c
+                SELECT Batch_Gift_Entry_Version__c,
+                        Form_Template__r.Template_JSON__c
                 FROM DataImportBatch__c
                 WHERE Id = :batchId
         ];
 
         if (batch.Batch_Gift_Entry_Version__c == null ||
                 batch.Batch_Gift_Entry_Version__c < 2.0) {
-            fieldNames = getDataImportFields(batchId, true);
+
+            fields = getCoreDataImportFields(includeRelationshipFields);
+            fields.addAll(getActiveFieldNamesFromBatch(batchId));
+
         } else if (batch.Batch_Gift_Entry_Version__c >= 2.0) {
-            // Lowercase the core fields list for use below to avoid duplication
-            for (String field : getCoreDataImportFields(true)) {
-                fieldNames.add(field.toLowerCase());
-            }
 
-            // Get all mapped source fields
-            // (no matter which template is used, it will be a subset of mapped fields)
-            // question: safe? or could a widget ever include fields that are not mapped?
-            for (SObjectField sObjectField :
-                    BDI_DataImportService.getDefaultMappingService()
-                            .getTargetFieldsBySourceField().keySet()) {
-                String field = String.valueOf(sObjectField).toLowerCase();
-                if (!fieldNames.contains(field)) {
-                    fieldNames.add(field);
-                }
+            for (String field : getCoreDataImportFields(includeRelationshipFields)) {
+                fields.add(field.toLowerCase());
             }
-        }
-        return fieldNames;
-    }
 
-    /*******************************************************************************************************
-    * @description Provides field mapping configuration used by BGE
-    */
-    public static Map<SObjectField, BDI_TargetFields> targetFieldsBySourceField = 
-        BDI_DataImportService.getDefaultMappingService().getTargetFieldsBySourceField();
+            // Must explicitly use BDI_MappingServiceAdvanced here (as opposed to
+            // BDI_DataImportService.getDefaultMappingService()) because of the need
+            // to reference the fieldMappingByDevName property.
+            Map<String,BDI_FieldMapping> fieldMappingByDevName =
+                    BDI_MappingServiceAdvanced.getInstance().fieldMappingByDevName;
+
+            FORM_Template t = (FORM_Template) JSON.deserialize(
+                    batch.Form_Template__r.Template_JSON__c,
+                    FORM_Template.class
+            );
+
+            for (FORM_Section formSection : t.layout.sections) {
```

Comment:
FYI @ptewson-sfdo this is the section where I swapped out collecting all mapped fields for getting all fields (and Name fields from related records) that are used in the Template.